File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/utils/StringUtil.java
package net.crystalapps.permission.runtime.library.utils;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

/**
 * Created by Syed Owais Ali on 10/14/2018.
 */
public class StringUtil {

    public static String parseMessage(@NonNull Context context, @StringRes int resId, @Nullable String defaultValue) {
        if (resId != -1) {
            try {
                return context.getResources().getString(resId);
            }
            catch (Exception ex) {
                return defaultValue;
            }
        }
        else {
            return defaultValue;
        }
    }
}

File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/utils/StringUtil.kt
package net.crystalapps.permission.runtime.library.utils

import android.content.Context
import androidx.annotation.StringRes

/**
 * Created by Syed Owais Ali on 10/14/2018.
 */
object StringUtil {

    fun parseMessage(context: Context, @StringRes resId: Int, defaultValue: String?): String? {
        return if (resId != -1) {
            try {
                context.resources.getString(resId)
            } catch (ex: Exception) {
                defaultValue
            }

        } else {
            defaultValue
        }
    }
}
